data/testdata/all_types.golden.json

```diff
@@ -134,9 +134,9 @@
       { "NaN": [0], "Inf": [1], "NegInf": [3] },
       { "NaN": [1], "Inf": [0], "NegInf": [3] },
       null,
-      { "NaN": [1, 3], "Inf": [4], "NegInf": [0] },
+      { "NaN": [3], "Inf": [4], "Undef": [1], "Nil": [2], "NegInf": [0] },
```

Comment:
@toddtreece we don't need to encode `Nil` here since `null` is a JSON-native type and is represented directly in the data. `entities` is for types that cannot need special handling after JSON decode (NaN, Inf, NegInf, undefined)

Reply:
i think i tried this originally, but must have missed a file and switched to `entities`. should be fixed in b9bb620